utils/AtlasExtractSubmesh.cpp

```diff
@@ -49,7 +49,7 @@ void copyNodeData(const atlas::Mesh& meshIn, const std::vector<int> keptNodeIndi
   auto ghostIn = atlas::array::make_view<int, 1>(nodesIn.ghost());
   auto flagsIn = atlas::array::make_view<int, 1>(nodesIn.flags());
 
-  const atlas::mesh::Nodes& nodes = meshOut.nodes();
+  atlas::mesh::Nodes& nodes = meshOut.nodes();
```

Comment:
otherwise the ArrayView below are ArrayView<const double,  x> which dont allow assignments